extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/handler/FSInternetRadioHandler.java

```diff
@@ -0,0 +1,232 @@
+                        // if radio is off, set all channels (except for 'POWER') to 'undefined'
+                        updateState(channel.getUID(), UnDefType.UNDEF);
+                    } else if (channel.isLinked()) {
+                        // update all channels that are linked
+                        switch (channel.getUID().getId()) {
+                            case CHANNEL_POWER:
+                                updateState(channel.getUID(), radioOn ? OnOffType.ON : OnOffType.OFF);
+                                break;
+                            case CHANNEL_VOLUME:
+                                updateState(channel.getUID(), DecimalType.valueOf(String.valueOf(radio.getVolume())));
+                                break;
+                            case CHANNEL_MODE:
+                                updateState(channel.getUID(), DecimalType.valueOf(String.valueOf(radio.getMode())));
+                                break;
+                            case CHANNEL_MUTE:
+                                updateState(channel.getUID(), radio.getMuted() ? OnOffType.ON : OnOffType.OFF);
+                                break;
+                            case CHANNEL_PRESET:
+                                // preset is write-only, ignore
+                                break;
+                            case CHANNEL_PLAY_INFO_NAME:
+                                updateState(channel.getUID(), StringType.valueOf(radio.getPlayInfoName()));
+                                break;
+                            case CHANNEL_PLAY_INFO_TEXT:
+                                updateState(channel.getUID(), StringType.valueOf(radio.getPlayInfoText()));
+                                break;
+                            default:
+                                logger.warn("Ignoring unknown channel during update: " + channel.getLabel());
+                        }
+                    }
+                }
+                updateStatus(ThingStatus.ONLINE); // set it back online, maybe it was offline before
+            } catch (Exception e) {
+                String msg = "Communication with radio (update) failed: " + e.getMessage();
+                logger.error(msg);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, msg);
+            }
+        }
+    };
+
+    public FSInternetRadioHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        // Long running initialization should be done asynchronously in background
+        radioLogin(true);
+
+        // also schedule a thread for polling with configured refresh rate
+        final BigDecimal period = (BigDecimal) getThing().getConfiguration().get(CONFIG_PROPERTY_REFRESH);
+        if (period != null && period.intValue() > 0) {
+            updateJob = scheduler.scheduleAtFixedRate(updateRunnable, period.intValue(), period.intValue(),
```

Comment:
WithFixedDelay is the better option as it does not queue up jobs (e.g. during debugging sessions).